### DISCUSSION TITLE
[@types/shell-escape] Missing ‘types’ in package.json

### DISCUSSION THREAD

**Discussion by esdmr** [DefinitelyTyped/DefinitelyTyped#61718] [Issues with a @types package] [answered]
The type for `shell-escape` has neither `types` nor `exports` in its [`package.json`](https://unpkg.com/browse/@types/shell-escape@0.2.0/package.json).

Without the `types` pointing to `index.d.ts`, typescript errs:

```
Could not find a declaration file for module 'shell-escape'. '/$DIR/node_modules/shell-escape/shell-escape.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/shell-escape` if it exists or add a new declaration (.d.ts) file containing `declare module 'shell-escape';`
```

Edit: The error persists with `@types/shell-escape` installed. It goes away if I manually add a `types` field in the `package.json` in `node_modules`.

[a comment by an automated account is not shown]

**Comment by BendingBender** [comment, marked as answer]
You should really read the error message from TypeScript which states:

>   Try `npm i --save-dev @types/shell-escape` if it exists or add a new declaration (.d.ts) file containing `declare module 'shell-escape';`

As it states, you need to install the `@types/shell-escape` package to get the TS definitions:
```sh
npm i --save-dev @types/shell-escape
```

or

```sh
yarn add --dev @types/shell-escape
```

**Reply by esdmr** [reply]
I should have specified that I have already installed `@types/shell-escape`, My bad. The error persists with it installed.

Also, the error goes away if I manually add a `types` field in the package.json in node_modules.

**Reply by BendingBender** [reply]
Just tried it out in a brand new project, everything works as it should. Without manually editing the `types` field.

**Reply by esdmr** [reply]
Oh. I have been bamboozled by VSCode. The error was only showing in VSCode, since it was using typescript beta instead of the stable workspace one. Switching the version seems to fix it. I should have tested it before reporting.

Thank you!
